### PR TITLE
Updated for game version 0.8.1

### PR DESCRIPTION
**Pull request by nickmcski**
Updated the harmony hooks to work with game version 0.8.1. I expect this will break compatibility with previous game versions.

### CODE REVIEW

**Comment by grasmanek94** [comment]
image
I think there's still something wrong with the food / item use amount.
Apart from that, nice work in updating it, thanks. When I find time I can update it, or maybe you can also update the PR to fix this.

**Comment by grasmanek94** [comment]
`public void OnVisit(NPCBase npc, out bool gotFood)` looks like the correct place, unfortunately it's one big glob of code, I'll try to come up with something.

File: StockpileStatisticsHooks.cs [hunk 1]
```diff
@@ -56,21 +56,21 @@ static void Postfix(Stockpile __instance, bool __result, ushort type, int amount
...
-	{
+	[HarmonyPatch(typeof(ColonyShopVisitTracker))]
+	[HarmonyPatch("OnVisit")]
+	public class ColonyShopVisitTrackerHookOnVisit
+    {
 		public  static Stockpile inFunctionStockpile = null;
 		public static bool isFood;
 
-		static void Prefix(Stockpile __instance, ref float currentFood, float desiredFoodAddition)
+		static void Prefix(ref NPCBase npc, ref bool gotFood)
```
Comment (grasmanek94) [inline StockpileStatisticsHooks.cs]: `npc` shouldn't be `ref` here, remove it.
gotFood.. I think we can make it `out`?

File: StockpileStatisticsHooks.cs [hunk 2]
```diff
@@ -56,21 +56,21 @@ static void Postfix(Stockpile __instance, bool __result, ushort type, int amount
...
 		{
 			// Log.WriteWarning("StockpileHookTryRemoveFood::Prefix");
-			inFunctionStockpile = __instance;
+			inFunctionStockpile = npc.Colony.Stockpile;
 			isFood = true;
 		}
 
-		static void Postfix(Stockpile __instance, bool __result, ref float currentFood, float desiredFoodAddition)
+		static void Postfix(ref NPCBase npc, ref bool gotFood)
```
Comment (grasmanek94) [inline StockpileStatisticsHooks.cs]: Remove `ref` from `npc`, `gotFood` = `out`?

File: modInfo.json [hunk 3]
```diff
@@ -1,23 +1,28 @@
...
+      {
+        "version": "0.0.2.0",
+        "notes": [
+          "Update for game version 0.8.1"
+        ]
+      }
+    ],
 		"dllpath": "grasmanek94.Statistics.dll"
 	}
 ]
```
Comment (grasmanek94) [inline modInfo.json:28]: If you want:
could you please add a contributors.txt and add yourself in the format that you want?

**Review by grasmanek94 — CHANGES_REQUESTED** [review]
a few small changes are also needed

**Comment by grasmanek94** [comment]
I was looking for the bug, and although I can't seem to find it at the moment, I did notice that we could add consumers for food with something like:

```
	[HarmonyPatch(typeof(ColonyShopVisitTracker))]
	[HarmonyPatch("OnVisit")]
	public class ColonyShopVisitTrackerHookOnVisit
    {
		public  static Stockpile inFunctionStockpile = null;
		public static bool isFood;
		public static NPCBase npc = null;

		static void Prefix(NPCBase arg_npc, bool gotFood)
		{
			// Log.WriteWarning("StockpileHookTryRemoveFood::Prefix");
			inFunctionStockpile = arg_npc.Colony.Stockpile;
			isFood = true;
			npc = arg_npc;
		}

		static void Postfix(NPCBase arg_npc, bool gotFood)
		{
			// Log.WriteWarning("StockpileHookTryRemoveFood::Postfix");
			inFunctionStockpile = null;
			isFood = false;
			npc = null;
		}
	}
// (...)
			if (ColonyShopVisitTrackerHookOnVisit.isFood)
			{
				AddConsumerRemoveProducer(itemStats, ColonyShopVisitTrackerHookOnVisit.npc, amount);

				itemStats.UseAsFood(amount);
			}
```

**Comment by grasmanek94** [comment]
I see WIP? Lemme know when to review :) @nickmcski

File: StockpileStatisticsHooks.cs (hunk 1 above)
Comment (nickmcski) [reply StockpileStatisticsHooks.cs]: I wasn't able to change gotFood to an out, it gave a compiler error as we aren't setting that value in our method. 

Would it be better to leave it as a ref, remove it, or is there a better way to get it as a out?

**Comment by nickmcski** [comment]
> I see WIP? Lemme know when to review :) @nickmcski

I've moved those WIP changes into a separate branch, I just kept running into issues and realized it likely requires a larger change to how we are storing data. I'll open a separate discussion for that. 

I think I've addressed all your comments and changes so I would be happy if you could take another look!

**Comment by grasmanek94** [comment]
We do still have a bug where some food isn't counted.. like foofoo :/ but hey, at least most of it works.

**Pull request merged**